openedx/adg/lms/applications/helpers.py

```diff
@@ -529,6 +541,71 @@ def get_application_hub_instructions(
     }
 
 
+def get_omnipreneurship_courses_instructions(courses):
+    """
+    Creates instructions for the Omnipreneurship courses depending on each course's status and whether that course has
+    a prerequisite or not
+
+    Arguments:
+        courses (list): List of courses for which instructions need to be made
+
+    Returns:
+        dict: Dictionary containing instructions for open, locked and unlocked courses
+    """
+    open_courses = locked_courses = unlocked_courses = ''
+
+    for course in courses:
+
+        if course['pre_req']:
+            if course['status'] == LOCKED:
+                locked_courses = f'{locked_courses}{_("Once you have completed")} {course["pre_req"]} ' \
+                                 f'{_("and received a passing grade, the")} ' \
+                                 f'{course["course"].display_name_with_default} {_("course will be unlocked. ")}'
+            else:
+                unlocked_courses = f'{unlocked_courses}{course["course"].display_name_with_default}, '
+
+        else:
+            open_courses = f'{open_courses}{course["course"].display_name_with_default}, '
+
+    if open_courses:
+        open_courses = replace_last_comma_occurrences(open_courses)
+
+        open_courses = f'{_("You will have to complete")} <em>{_("all")}</em> ' \
+                       f'{_("of the Omnipreneurship courses. However, you will have to enroll in and")} ' \
+                       f'<em>{_("complete")} {open_courses} {_("first.")}</em>'
+
+    if unlocked_courses:
+        unlocked_courses = replace_last_comma_occurrences(unlocked_courses)
+
+        verb = 'have' if 'and' in unlocked_courses else 'has'
+
+        unlocked_courses = f'{unlocked_courses} {_(verb)} {_("been unlocked for you to enroll in and complete! ")}'
+
+    return {
+        'open_courses': open_courses,
+        'locked_courses': locked_courses,
+        'unlocked_courses': unlocked_courses
+    }
+
+
+def replace_last_comma_occurrences(string):
+    """
+    Removes trailing comma and replaces the replaces the last remaining comma (if any) with and
+
+    Arguments:
+        string (str): the string from which the comma(s) are need to be removed
+
+    Returns:
+        str: a string with removed comma(s)
+    """
+    string = string[:-2]
```

Comment:
Why are we removing the last two characters from the string? Maybe adding a comment would help?